keymanager/src/secrets/interface.rs

```diff
@@ -0,0 +1,17 @@
+use oasis_core_runtime::consensus::beacon::EpochTime;
+
+use crate::crypto::{Secret, VerifiableSecret};
+
+/// Interface for providing master and ephemeral secrets.
+pub trait SecretProvider {
+    /// Returns an iterator that provides access to all the replicas of the master secret
+    /// for the given generation.
+    fn master_secret_iter(
+        &self,
+        generation: u64,
+    ) -> Box<dyn Iterator<Item = VerifiableSecret> + '_>;
```

Comment:
Do these need to be boxed? You could use (and similar below):
```suggestion
    ) -> impl Iterator<Item = VerifiableSecret> + '_;
```

Reply:
I used my solution to avoid the following error:
```
`impl Trait` only allowed in function and inherent method return types, not in trait method return
see issue #91611 <https://github.com/rust-lang/rust/issues/91611> for more information
add `#![feature(return_position_impl_trait_in_trait)]` to the crate attributes to enable
``

Reply:
Ah right, not yet stable in traits.